biom/tests/test_cli/test_add_metadata.py

```diff
@@ -20,10 +21,11 @@ class TestAddMetadata(TestCase):
     def setUp(self):
         """Set up data for use in unit tests."""
         self.cmd = _add_metadata
-        with tempfile.NamedTemporaryFile('w') as fh:
+        with tempfile.NamedTemporaryFile('w', delete=False) as fh:
             fh.write(biom1)
             fh.flush()
             self.biom_table1 = biom.load_table(fh.name)
+        os.unlink(fh.name)
```

Comment:
Could `self.temporary_fh_name` be set within the scope of the context manager, and move the `os.unlink` to `tearDown`? The reason for the former is the current approach relies on scope creep which works but borders on surprising behavior. The latter is important as, if a test fails, cleanup will only reliably occur if the `unlink` is performed by `tearDown`.